cognite/client/utils/_token_generator.py

```diff
@@ -1,11 +1,14 @@
 import datetime
+import threading
 from typing import Dict, List, Optional
 
 from oauthlib.oauth2 import BackendApplicationClient, OAuth2Error
 from requests_oauthlib import OAuth2Session
 
 from cognite.client.exceptions import CogniteAPIKeyError
 
+LOCK = threading.Lock()
```

Comment:
Shouldn't this be a reenter lock such that the same thread can acquire it form itself?
```suggestion
LOCK = threading.RLock()
```

Reply:
Hmm, not sure understand how/why that would be needed? 🤔  Afaik, each thread is told to make one request, and while building the headers it makes one call to this non-recursive method to get a valid token.